File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    priority_queue<int,vector<int>,greater<int>> min;
    int k;
    KthLargest(int k, vector<int>& nums) {
        
        this->k=k;
        for(int &num:nums){
            min.push(num);
            if(min.size()>k) min.pop();
        }
        
    }
    
    int add(int val) {
        min.push(val);
        if(min.size()>k) min.pop();
        return min.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.py
class KthLargest(object):

    def __init__(self, k, nums):
        self.minHeap=nums
        self.k=k
        heapq.heapify(self.minHeap)
        while len(self.minHeap)>k:
            heapq.heappop(self.minHeap)    
        

    def add(self, val):
        heapq.heappush(self.minHeap,val)
        if len(self.minHeap)>self.k:
            heapq.heappop(self.minHeap)
        return self.minHeap[0]

        
        

# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)